src/navigators/RootNavigator.js

```diff
@@ -1,23 +1,108 @@
-import React from 'react';
+import React, { useRef, useState, useEffect } from 'react';
+import { useSelector } from 'react-redux';
 import { NavigationContainer } from '@react-navigation/native';
 import { createStackNavigator } from '@react-navigation/stack';
-import { TabNavigator } from './';
-import { Onboarding, SearchResults, ErrorScreen } from 'screens';
-import { ROUTES } from 'constants';
+import { TabNavigator } from './TabNavigator';
+import {
+  Onboarding,
+  SearchResults,
+  PropertyScreen,
+  ErrorScreen,
+} from 'screens';
+import { SizeButton } from 'components';
+import { COLORS, ROUTES } from 'constants';
 
 const Stack = createStackNavigator();
 
-export const RootNavigation = () => (
-  <NavigationContainer>
-    <Stack.Navigator
-      screenOptions={{
-        headerShown: false,
-      }}
-    >
-      <Stack.Screen name={ROUTES.ONBOARDING} component={Onboarding} />
-      <Stack.Screen name={ROUTES.TAB_NAVIGATOR} component={TabNavigator} />
-      <Stack.Screen name={ROUTES.SEARCH_RESULTS} component={SearchResults} />
-      <Stack.Screen name={ROUTES.ERROR_SCREEN} component={ErrorScreen} />
-    </Stack.Navigator>
-  </NavigationContainer>
-);
+const { Navigator, Screen } = Stack;
+
+const headerStyles = {
+  headerStyle: {
+    backgroundColor: COLORS.FLAMINGO,
+    shadowColor: COLORS.BLACK,
+    shadowOffset: {
+      width: 0,
+      height: 4,
+    },
+    shadowOpacity: 0.5,
+    shadowRadius: 12,
+    elevation: 5,
+  },
+  headerTintColor: COLORS.WHITE,
+  headerBackTitle: ' ',
+};
+
+export const RootNavigation = ({ alreadyLaunched }) => {
+  const error = useSelector(state => state.error);
+  const ref = useRef();
+
+  useEffect(() => {
+    if (error.message) {
+      ref.current.navigate(ROUTES.ERROR_SCREEN);
+    }
+  }, [error.message]);
+
+  useEffect(() => {
+    if (alreadyLaunched) {
+      ref.current.navigate(ROUTES.TAB_NAVIGATOR);
+    }
+  }, [alreadyLaunched]);
```

Comment:
You can remove it if you pass prop initialRouteName (the name of the route to render on first load of the navigator.)
https://reactnavigation.org/docs/stack-navigator